extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/serde/CommandSpec.scala

```diff
@@ -71,4 +79,33 @@ case class TableCommandSpec(
     classname: String,
     tableDescs: Seq[TableDesc],
     opType: String = "QUERY",
-    queryDescs: Seq[QueryDesc] = Nil) extends CommandSpec
+    queryDescs: Seq[QueryDesc] = Nil) extends CommandSpec {
+  def queries: LogicalPlan => Seq[LogicalPlan] = plan => {
+    queryDescs.flatMap { qd =>
+      try {
+        qd.extract(plan)
+      } catch {
+        case e: Exception =>
+          LOG.warn(qd.error(plan, e))
+          None
+      }
+    }
+  }
+}
+
+case class ScanSpec(
+    classname: String,
+    scanDescs: Seq[ScanDesc]) extends CommandSpec {
+  override def opType: String = "QUERY"
```

Comment:
maybe extract this repeated opType "QUERY" to the object for preventing magic value? or use `OperationType.QUERY.toString`.